sdk/src/wallet/utxo.ts

```diff
@@ -17,6 +18,69 @@ export const getBtcNetwork = (name: BitcoinNetworkName) => {
 
 type Output = { address: string; amount: bigint } | { script: Uint8Array; amount: bigint };
 
+class TreeNode<T> {
+    val: T;
+    children: TreeNode<T>[];
+
+    constructor(val: T, children: TreeNode<T>[] = []) {
+        this.val = val;
+        this.children = children;
+    }
+}
+
+const createUtxoNodes = (utxos: UTXO[], cardinalOutputsSet: Set<string>) =>
+    utxos.reduce(
+        (acc, utxo) => {
+            if (!cardinalOutputsSet.has(OutPoint.toString(utxo)))
+                acc.push(new TreeNode<OutputNodeData>({ ...utxo, cardinal: true }));
+            else acc.push(null);
+
+            return acc;
+        },
+        [] as (TreeNode<OutputNodeData> | null)[]
+    );
+
+const processNodes = async (rootNodes: (TreeNode<OutputNodeData> | null)[], esploraClient: EsploraClient) => {
+    const queue = Array.from(rootNodes);
+
+    while (queue.length > 0) {
+        const childNode = queue.shift();
+
+        if (childNode === null) continue;
+
+        const txInscriptions = await getTxInscriptions(esploraClient, childNode.val.txid);
+
+        if (txInscriptions.length === 0) {
+            const transaction = await esploraClient.getTransaction(childNode.val.txid);
+
+            // if not confirmed check inputs for current utxo
+            if (!transaction.status.confirmed) {
```

Comment:
Since we don't ever check Ord again the mempool tx might be spending an ordinal, you only check the confirmation here.

Reply:
fixed